mongomock/filtering.py

```diff
@@ -214,6 +214,9 @@ def iter_key_candidates(key, doc):
 
     if doc is None:
         return ()
+    
+    if isinstance(doc, DBRef) and key == '$id':
```

Comment:
I can suggest to make this check `DBRef and isinstance(doc, DBRef)` and instead of using `doc.id` directly use  more general `return [doc.as_doc().get(key, NOTHING)]`. This will align well with same code from #878

Reply:
Can the `DBRef` class itself be a `None`, or why check against it?

I like the more general return though, works perfectly on my tests.